neurodamus/core/configuration.py

```diff
@@ -944,6 +945,18 @@ def _keep_coreneuron_data(config: _SimConfig):
     log_verbose("delete_corenrn_data = %s", config.delete_corenrn_data)
 
 
+@SimConfig.validator
+def _report_buffer_size(config: _SimConfig):
```

Comment:
In the case of `--restore`, we will reuse the `report.conf` from save, so `--report-buffer-size` will not work.  Shall we add a warning about that ?

Reply:
I'll simply add the warning now. Changing the buffer size after a restore is not a priority and I can imagine it is not s o straight forward

Reply:
If needed, probably should be updated here.
https://github.com/openbraininstitute/neurodamus/blob/387307511431a93fbb29c86bab153bddd0b75973/neurodamus/core/coreneuron_configuration.py#L131
It is not urgent as it is only us that play with the buffer size at the moment. I will let you decide @jamesgking .

Reply:
that usually deals with the updated tstop. Maybe you can add a check there. I think the safest would be to complain before: if you specify `buffer-size` and `restore` you get an error. I would not use a warning as they are rarely seen

Reply:
>that usually deals with the updated tstop.

The function name and the docstring indicate that it could update other parameters in the report.conf, e.g. buffer_size. Right? 

In my previous comment, I meant if we need to update the buffer_size in restore, probably it is in this function.

Reply:
> In my previous comment, I meant if we need to update the buffer_size in restore, probably it is in this function.

ah I did not understand. I thought it was about the warning since Jim mentioned it just before. 

btw, I am planning to change the way we deal with report.conf because it is a little clunky and prone to errors

Reply:
Added Configuration Error when both are present